### ISSUE TITLE
There is a bug with latest stable torch version and the following Nightly versions related to `optimize_for_mobile`

### ISSUE THREAD

**Issue by Shahrullo** [pytorch/pytorch#76807]
### 🐛 Describe the bug

There is a bug in `optimize_for_mobile` with latest torch version and the following nightly versions. Please look at this [discussion](https://discuss.pytorch.org/t/torch-utils-mobile-optimizer-optimize-for-mobile-is-resulting-different-output-than-torch-model-and-jit-model/150183) for full details.

**The short problem description:** the model is giving different result after `optimize_for_mobile`

The problem started  with stable torch version 1.11.0 and the following Nightly versions. But it is working correctly with previous torch versions. 

Here is the minimal code snippet to reproduce the error:

```
import torch
import torchvision
from torch.utils.mobile_optimizer import optimize_for_mobile

model = torchvision.models.resnet50(pretrained=True)
model.eval()


out1 = model(torch.ones([1, 3, 224, 224], dtype=torch.float))
out2 = traced_script_module(torch.ones([1, 3, 224, 224], dtype=torch.float))
out3 = traced_script_module_optimized(torch.ones([1, 3, 224, 224], dtype=torch.float))
```
and results:

```
tensor([[-3.0806e-01,  7.9845e-02, -1.1900e+00, -1.4837e+00, -5.1359e-01,...]])
tensor([[-3.0806e-01,  7.9845e-02, -1.1900e+00, -1.4837e+00, -5.1359e-01,...]])
tensor([[-5.2194e+00,  1.4939e+00, -2.2703e+00, -5.4212e+00, -2.0551e+00,...]])
```

### Versions

 torch version 1.11.0 and the following Nightly versions.

**Comment by Shahrullo** [comment]
@Gaozhongpai I think something went wrong after torch 1.11.0 version. Try earlier stable versions, like torch `1.10.1`, it outputs the same results

**Comment by nsriniva03** [comment]
I have the same issue with torch version 1.12.0. Using torch version 1.10 fixed it. Any updates on this?

**Comment by Shahrullo** [comment]
@nsriniva03 I don't think so. The bug is still the issue with new versions. It's better to use earlier  version currently

**Comment by gurkirt** [comment]
Has this been fixed in `1.13.0` ?

**Comment by AkiSakurai** [comment]
I can confirm the bug still exists in PyTorch 2.2. Furthermore, I discovered a workaround: if you manually fuse the batch normalization layer before passing the module to optimize_for_mobile, the issue gets resolved.


``` python
import torch
import torchvision
from torch.utils.mobile_optimizer import optimize_for_mobile
from torch.ao.quantization.quantize_jit import fuse_conv_bn_jit

model = torchvision.models.resnet50(weights=torchvision.models.ResNet50_Weights.DEFAULT)
model.eval()
example = torch.rand(1, 3, 224, 224)

traced_module = torch.jit.trace(model, example, strict=True)
traced_module_fused = fuse_conv_bn_jit(traced_module)
traced_module_optimized = optimize_for_mobile(traced_module)
traced_module_fused_optimized = optimize_for_mobile(traced_module_fused)

data = torch.ones([1, 3, 224, 224], dtype=torch.float)

models = {
     "model" : model, 
     "traced_module" : traced_module, 
     "traced_module_optimized" : traced_module_optimized, 
     "traced_module_fused_optimized" : traced_module_fused_optimized
}

for k, v in models.items():
    print(k, v(data)[0,0])
    
#model tensor(-0.7476, grad_fn=<SelectBackward0>)
#traced_module tensor(-0.7476, grad_fn=<SelectBackward0>)
#traced_module_optimized tensor(0.0048)
#traced_module_fused_optimized tensor(-0.7476)

```
